tools/gfx/persistent-shader-cache.cpp

```diff
@@ -121,13 +161,24 @@ void PersistentShaderCache::addEntry(const slang::Digest& dependencyDigest, cons
         deleteLRUEntry();
     }
 
-    ShaderCacheEntry entry = { dependencyDigest, astDigest };
+    auto lastAccessedTime = duration_cast<duration<double>>(high_resolution_clock::now().time_since_epoch()).count();
+
+    ShaderCacheEntry entry = { dependencyDigest, contentsDigest, lastAccessedTime };
     auto entryNode = entries.AddFirst(entry);
     keyToEntry.Add(dependencyDigest, entryNode);
 
     mutableShaderCacheFileSystem->saveFileBlob(DigestUtil::toString(dependencyDigest).getBuffer(), compiledCode);
 
-    saveCacheToFile();
+    indexStream.seek(SeekOrigin::End, 0);
+    entryToOffset.Add(entry, indexStream.getPosition());
```

Comment:
For binary files, just seek and write the raw bytes.

Reply:
done